vault-upgrade-hook/src/main/java/biz/netcentric/vlt/upgrade/handler/SlingUtils.java

```diff
@@ -35,12 +35,7 @@ public ResourceResolver getResourceResolver(Session session) {
     }
 
     public boolean hasRunModes(Set<String> requiredRunModes) {
-        Set<String> actualRunMode = getRunModes();
-        boolean hasAll = true;
-        for (Iterator<String> it = requiredRunModes.iterator(); it.hasNext() && hasAll; ) {
-            hasAll = actualRunMode.contains(it.next());
-        }
-        return hasAll;
+        return requiredRunModes.isEmpty() || getRunModes().containsAll(requiredRunModes);
```

Comment:
IMHO the first condition is not even necessary

Reply:
You are right, done and added a test case.